scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1022,12 +1050,14 @@ def update_cache( self, run_state ):
             try:
                 # we only pre warm the pod cache and the cluster name
                 # controllers are cached on an as needed basis
-                if local_state.batch_pod_updates:
-                    self._pods.update( local_state.k8s, local_state.node_filter, 'Pod' )
+                if local_state.batch_pod_updates and not self._state.cache_config.use_controlled_warmer:
+                    self._pods_cache.update(local_state.k8s, local_state.node_filter, 'Pod')
                 self._update_cluster_name( local_state.k8s )
-                self._update_api_server_version(local_state.k8s)
-
-                runtime = self._get_runtime( local_state.k8s )
+                if not self._state.cache_config.use_controlled_warmer:
+                    self._update_api_server_version(local_state.k8s)
+                    runtime = self._get_runtime( local_state.k8s )
```

Comment:
Avoid checking runtime otherwise it is not rate limited.  Also for now, warming cache => docker.